adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java

```diff
@@ -433,6 +443,24 @@ public Schema getTableSchema() {
         Object o = objectEnumeration.current();
         final TKey key = keySelector.apply(o);
 
+        if (sortedPrefixKeySelector != null) {
+          final Object sortedKey = sortedPrefixKeySelector.apply(o);
+          // If sortedPrefixKeySelector is not null, we can only stop reading rows when
+          // the
+          // sorted key prefix changes and we have enough unique groups since the rows
+          // have to be
+          // sorted on the client we have to read all the rows that have the same sorted
+          // primary key prefix)
+          if (lastSortedKey != null && !sortedKey.equals(lastSortedKey) && uniqueGroupCount > groupFetchLimit) {
```

Comment:
wouldn't `lastSortedKey` will always be null during the first iteration from line 424?